suitcase/tiff_series/__init__.py

```diff
@@ -160,6 +161,13 @@ class Serializer(tiff_stack.Serializer):
     **kwargs : kwargs
         kwargs to be passed to ``tifffile.TiffWriter.save``.
     """
+    def __init__(self, directory, file_prefix='{uid}-', bigtiff=False,
+                 byteorder=None, imagej=False, **kwargs):
+        super().__init__(directory, file_prefix, bigtiff,
+                         byteorder, imagej, **kwargs)
+        # maps stream name to dict that map field name to index (#)
+        self._counter = defaultdict(lambda: defaultdict(lambda: 0))
```

Comment:
use `itertools.count`?

Reply:
@tacaswell I am not sure that in this case `itertools.count` works. mainly because this is not just an `int` but many `int`s each relating to separate streams and fields.  Moreover it is not _always_ updated in a loop, and is only increased _if_ in this doc we see a 2D array for a given `streamname` and `field`. So I am not sure exactly how to use `itertools.count` in this case.

Reply:
@tacaswell is suggesting one `itertools.count` instance per entry:

`lambda: 0` -> `itertools.count`.

Then:

```py
self._counter[streamname][field] += 1                           
num = self._counter[streamname][field]
```

becomes

```py
num = next(self._counter[streamname][field])
```

Slightly more succinct and less flexible than an integer (in a good way, in the sense of one less place for something unexpected to happen).

Reply:
thanks @danielballan I wasn't aware of the option of using `next(itertools.count)` which was the piece of the solution I was missing. I will fix this later today.

Reply:
fixed